### PR TITLE
Feature/2020 07 10 review

### LINKED ISSUE #30: Decide on guidelines for making changes to FOM modules
@lofstrandbjorn has asked me to draft some guidelines for working with GitHub to make changes and releases, etc. In short, a workflow.

My experience from software development is that such a workflow can be a tremendous help in the long run, if kept as simple as possible. It should be a help in keeping things organized, not a hindrance.

Below is my first draft of a workflow for making edits to FOM modules. This needs some feedback, in particular the question of who/how many different parties should review each change. Is it enough that one party outside the editor's organization reviews the change? Should we somehow mandate what party reviews which changes, or should we leave it open? Should a particular module always be reviewed by the same party/parties, or should we encourage parties to review many different modules over time?

Of course, the higher-level discussions already happen in Issues: reviews are lower-level and meant to catch more detailed issues and errors, meaning it's probably not necessary for more than one or two parties to review each Pull Request. Looking forward to hearing your thoughts on this.

By the way, what Git clients are you using? I prefer Git bash myself, but I've evaluated a lot of GUI clients. In case you're not happy with your current client, I can recommend Git Gui (included with Git distribution on Windows) or GitHub Desktop. Many other GUI Git clients I've seen, like TortoiseGit and SourceTree, introduce a lot of unnecessary confusion.

---

> ## Making changes
> All changes should be initiated by creating a GitHub Issue. Issues may also be created for the purpose of discussing possible future changes without suggesting a concrete plan of action.
> 
> The develop branch should at all times contain only changes that have been reviewed and approved by at least one party that was not directly involved in making them. To this end, we adapt a workflow involving edit branches, pull requests and pull request reviews.
> 
> The workflow is closely based the commonly used “Gitflow” workflow, originally presented [here](https://nvie.com/posts/a-successful-git-branching-model/).
> 
> ### Creating a Pull Request
> 
> When ready to make a change, create a branch based on the develop branch. Name the branch on the form “edit/...”, and include the number of the issue where the changes are discussed in the branch name.
> 
> An edit branch may refer to more than one Issue, in which case its name should preferably include the numbers of all of them. This is not mandatory, however.
> 
> Make the changes on the branch, and push the branch to GitHub.
> 
> When you think the changes are ready, create a Pull Request, pointing it to the develop branch. Add reviewers to the PR as appropriate. Link the PR to each Issue it addresses, if it has not been automatically linked already.
> 
> ### Reviewing a Pull Request
> 
> When reviewing a pull request, comment on individual changes in the “Files changed” view. Then click the button marked “Review changes”, add any additional comments not related to individual changes, select Comment, Approve or Request changes, and submit the review.
> 
> ### Merging a Pull Request
> 
> Any edits that happen as a result of discussions and reviews of the PR should be committed and pushed to the same edit branch, which will automatically update the Pull Request.
> 
> The PR must be reviewed and approved by at least one party from a different organization than the creator of the PR before it can be merged to the develop branch.
> 
> If discussions of a PR raises questions that have not been addressed in a GitHub Issue, they should be added to a new or existing Issue and agreed upon before the PR is approved or merged.

### LINKED ISSUE #31: Update NETN-FOM module table description in documentation
Reflect changes made from merging Aggregate with MRM and add descriptions of each module

### PR DESCRIPTION
**Pull request by lofstrandbjorn**
CLOSES #30 
CLOSES #31

### CODE REVIEW

File: flow.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Work-flow
+
+## Repositories
+| Type | Description |
+| ---| ---| ---| 
+| NETN FOM  | Main Repository for the entire FOM including all related FOM Modules.|
```
Comment (linusonline) [inline flow.md:6]: The name of the repo is actually NETN-FOM. Also, looking at this file from regular GitHub UI, there seems to be some markdown problems in this table.

File: flow.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+| NETN-* | Repository for a specific FOM Module.|
+
+## Branches
+
+| Type | Description |
+| ---| ---|
+| Master Branch |The latest officially released version. |
+| Develop Branch |The lastest development version with reviewed changes. |
+| Release Branches |Created to prepare for new official release. |
```
Comment (linusonline) [inline flow.md:15]: Suggest: "Temporary, used to prepare for a release."

File: flow.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+| Maintenance Branches |Opened to maintain specific baseline or older version version. |
```
Comment (linusonline) [inline flow.md:16]: strike last "version". Also, move above release branches, since maintenance branches are permanent.

File: flow.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+| Edit Branches |Openend to develop proposed updates and closed after pull-request review and approval.|
```
Comment (linusonline) [inline flow.md:17]: Suggest: "Temporary, used to develop..."

File: flow.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+If discussions of a PR raises questions that have not been addressed in a GitHub Issue, they should be added to a new or existing Issue and agreed upon before the PR is approved or merged.
+
+## Releases
+### Making a release of a single module
+1. Create a release branch based on the develop branch, with a name on the form “release/...”, including the intended name of the release in the branch name.
+2. Make any final, pre-RC changes on the release branch, such as setting correct meta data, updating the changelog or adding documentation. Remember this includes setting the version name to the upcoming release or release candidate version wherever it is mention inside the repository, such as the FOM module identification table.
+3. Create a tag at the commit intended as the release candidate. Name it <version-name>-rcN, where N is the sequential number of the RC for this version, starting at 1.
+4. Optional: Make a release.
+5. Perform review activities.
```
Comment (linusonline) [inline flow.md:51]: Missing an empty line after this line.

Comment (lofstrandbjorn) [reply flow.md:51]: Fixed as suggested.

File: flow.md (hunk 4 above)
Comment (lofstrandbjorn) [reply flow.md]: Fixed as suggested

File: flow.md (hunk 3 above)
Comment (lofstrandbjorn) [reply flow.md]: Fixed as suggested

File: flow.md (hunk 2 above)
Comment (lofstrandbjorn) [reply flow.md]: Fixed as suggested

File: flow.md (hunk 1 above)
Comment (lofstrandbjorn) [reply flow.md]: Fixed.

File: NETN-FOM.md [hunk 6]
```diff
@@ -16,21 +16,20 @@ The NETN FOM provides standard interfaces for the representation of simulated en
...
-|NETN-ETR| | |
-|NETN-ORG| | |
-|NETN-SE| | |
-|NETN-COM| | |
+|Module|Version|Dependency|Description|
+|---|---|---|---|
+|NETN-BASE| v2.0 |RPR-Base | Common definitions of datatypes and extends the RPR-BASE FOM Module.|
+|NETN-Physical| v2.0 |NETN-BASE, RPR-Physical|Representation of Physical Entities in a federated distributed simulation. |
+|NETN-MRM| v2.0 |NETN-BASE, RPR-Aggregate | Aggregate level entity simulation, aggregation and disaggregation of units. Division and merging of unit resources. |
+|NETN-COM| v1.0 |NETN-BASE, RPR-Communication| Representation of Communication Networks and the status och communication links.|
```
Comment (sam-d-hall) [inline NETN-FOM.md:24]: "... and the status **of** communication links."

Comment (lofstrandbjorn) [reply NETN-FOM.md:24]: fixed

**Pull request merged**